packages/application-shell/src/components/application-shell/redirect-to-login.tsx

```diff
@@ -81,9 +86,13 @@ const RedirectToLogin = () => {
       />
     );
   }
+
+  const mcOrigin = servedByProxy ? getMcOrigin(window.app.mcApiUrl) : undefined;
```

Comment:
I would consider moving this into a hook as `useMcOrigin` to be able to memoize the computation a bit using a `useMemo`. We're now creating URL objects on each render. As the resulting value is primitive if will have a stable reference but I wonder if it could make sense to avoid obviously needless computations here as the `window.app.mcApiUrl` won't change.

Reply:
To elaborate:

```js
function useMcOrigin() {
   const isServedByProxy = useIsServerdByProxy();

   return useMemo(() => servedByProxy ? getMcOrigin(window.app.mcApiUrl) : undefined, [window.app.mcApiUrl, servedByProxy]);
}
```

I like keeping computation in a small getter as it's easier to test and allow logic to be lifted into different contexts.

Reply:
Ok. Sounds good as long as we don't create 100s or URL objects.